tools/crm_ticket.c

```diff
@@ -577,24 +543,38 @@ main(int argc, char **argv)
             goto done;
         }
 
-        if (!allow_modification(options.ticket_id)) {
-            exit_code = CRM_EX_INSUFFICIENT_PRIV;
-            g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
-                        "Ticket modification not allowed");
-            goto done;
-        }
-
         if (attr_delete != NULL) {
             rc = pcmk__ticket_remove_attr(out, cib_conn, scheduler, options.ticket_id,
-                                          attr_delete);
+                                          attr_delete, options.force);
+
+            if (rc == EACCES) {
+                ticket_revoke_warning(options.ticket_id);
+                g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
+                            "Ticket modification not allowed without --force");
+            }
         } else {
             rc = pcmk__ticket_set_attr(out, cib_conn, scheduler, options.ticket_id,
-                                       attr_set);
+                                       attr_set, options.force);
+
+            if (rc == EACCES) {
+                const char *value = NULL;
+
+                g_hash_table_lookup_extended(attr_set, PCMK__XA_GRANTED, NULL,
+                                             (gpointer *) &value);
```

Comment:
I don't think this ever needed extended, we're not using the return value, passing a NULL key, or getting the original key